main.go

```diff
@@ -243,63 +238,102 @@ func main() {
 					log.Fatal("unexpected method type")
 				}
 			}
+
+			if err := wr.Flush(); err != nil {
```

Comment:
Why this over the defers? IIRC the `log.Fatal` actually causes the program to exit w/ an error?

Reply:
The linter of my IDE complained that calling `defer` in a loop can lead to resource leaks. Also I think we want to catch errors that happen here and not just silently move on if flushing or closing fails.